shapely/geometry/base.py

```diff
@@ -917,6 +933,18 @@ def geoms(self):
     def __bool__(self):
         return self.is_empty is False
 
+    def __eq__(self, other):
+        if not isinstance(other, BaseGeometry):
+            return NotImplemented
+        return (
+            type(other) == type(self)
+            and len(self.geoms) == len(other.geoms)
+            and all(x == y for x, y in zip(self.geoms, other.geoms))
```

Comment:
Use binary geometry aliases "a" and "b", instead of coordinate dimension names.
```suggestion
            and all(a == b for a, b in zip(self.geoms, other.geoms))
```